compiler/rustc_parse/src/parser/item.rs

```diff
@@ -1216,6 +1216,24 @@ impl<'a> Parser<'a> {
 
     /// Parses an enum declaration.
     fn parse_item_enum(&mut self) -> PResult<'a, ItemInfo> {
+        if self.prev_token.is_keyword(kw::Enum) && self.token.is_keyword(kw::Struct) {
```

Comment:
We don't need to check prev token

```suggestion
        if self.token.is_keyword(kw::Struct) {
```

Reply:
Okay, because that happened before calling `parse_item_enum`.